File: Gdx/FellDungeon/app/src/main/java/com/dugsolutions/fell/map/gen2/GenZConstant.java
package com.dugsolutions.fell.map.gen2;

/**
 * Generate a constant height over all values. Can be sub-classed for generators
 * that have at least a bounds and a height.
 */
public class GenZConstant extends GenZBase {
	protected float mHeight;

	public GenZConstant(float height) {
		super();
		mHeight = height;
	}
	
	public GenZConstant(int count, float height) {
		super(count);
		mHeight = height;
	}

	public float getHeight() {
		return mHeight;
	}

	@Override
	protected void setZ(int x, int y) {		
		icalc.addHeight(x, y, mHeight);
	}

}

File: Gdx/FellDungeon/app/src/main/java/com/dugsolutions/fell/map/gen2/GenZConstant.kt
package com.dugsolutions.fell.map.gen2

/**
 * Generate a constant height over all values. Can be sub-classed for generators
 * that have at least a bounds and a height.
 */
open class GenZConstant : GenZBase {
    var height: Float
        protected set

    constructor(height: Float) : super() {
        this.height = height
    }

    constructor(count: Int, height: Float) : super(count) {
        this.height = height
    }

    override fun setZ(x: Int, y: Int) {
        icalc!!.addHeight(x, y, height)
    }
}